### PR TITLE
update swappable-obj-proxy to 2.2.0

### PR DESCRIPTION
**Pull request by BelfordZ**
Network controller and selected network controller are the two places updated by this PR.

[See here for more details](https://github.com/MetaMask/swappable-obj-proxy/pull/53)

not required for https://github.com/MetaMask/metamask-extension/pull/22202 but updating this here is a matter of good package hygene. 

### `@metamask/selected-network-controller`

- **FIXED**: Event listeners added to the blockTracker via a different proxy are migrated undesireably.

### `@metamask/package-b`

- **FIXED**: Event listeners added to the blockTracker via a different proxy (or directly on the blockTracker) are migrated undesireably.

## Checklist

- [ ] I've updated the test suite for new or updated code as appropriate
- [ ] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [ ] I've highlighted breaking changes using the "BREAKING" category above as appropriate

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -2966,6 +2966,13 @@ __metadata:
   languageName: node
   linkType: hard
 
+"@metamask/swappable-obj-proxy@npm:^2.2.0":
```
Comment (Gudahtt) [inline yarn.lock:2969]: Could you run `yarn dedupe` to get rid of the reference to the old version? This is in-range of `^2.1.0` still, which is what's keeping the `2.1.0` around in the lockfile

**Comment by Gudahtt** [comment]
I don't think this blocks the extension PR, as this is an in-range update. We can update this package in the extension with no change to the controllers. But, doesn't hurt to bump it here too

**Comment by BelfordZ** [comment]
> I don't think this blocks the extension PR, as this is an in-range update (`2.2.0` still satisfies the requested range `^2.1.0`). We can update this package in the extension with no change to the controllers. But, doesn't hurt to bump it here too

yep you are correct there - its not blocking and the other PR can move forward without. I will fix the wording on that.

**Comment by BelfordZ** [comment]
hmm still doesnt like that... odd

**Review by mcmire — APPROVED** [review]
Looks good!

**Pull request merged**